### ISSUE TITLE
trace view ...

### ISSUE THREAD

**Issue by goessner** [goessner/mec2#7]
... implemented and partially tested.

![grafik](https://user-images.githubusercontent.com/4902746/43968610-f55a0444-9cc7-11e8-8507-e61c149ea9ec.png)
```js
{
    id: '4bar',
    gravity: false,
    nodes: [
        {id:'A0',x:100,y:100,base:true, idloc:'sw'},
        {id:'A',x:100,y:150, idloc:'nw'},
        {id:'B',x:350,y:220, idloc:'ne'},
        {id:'C',x:250,y:250},
        {id:'B0',x:300,y:100,base:true, idloc:'se'},
    ],
    constraints: [
        { id:'a',p1:'A0',p2:'A',len:{type:'const'},ori:{type:'drive',Dt:3} },
        { id:'b',p1:'A', p2:'B',len:{type:'const'} },
        { id:'c',p1:'B0', p2:'B',len:{type:'const'} },
        { id:'d',p1:'B', p2:'C',len:{type:'const'},ori:{type:'ref',ref:'b'} }
    ],
    views: [
        { id:'path',type:'trace',p:'C',Dt:3, fill:'orange' }
    ]
}
```
![grafik](https://user-images.githubusercontent.com/4902746/43968770-60d95ef4-9cc8-11e8-83dc-b1a276c255c0.png)
```js
{
    id: 'double-pendulum',
    gravity: false,
    nodes: [
        {id:'A0',x:250,y:250,base:true},
        {id:'A1',x:250,y:150},
        {id:'A2',x:250,y: 50},
    ],
    constraints: [
        { id:'a1',p1:'A0',p2:'A1',len:{type:'const'} },
        { id:'a2',p1:'A1',p2:'A2',len:{type:'const'} },
    ],
    views: [
        { id:'aly',type:'trace',p:'A2',Dt:2 }
    ]
}
```

**Comment by jauhl** [comment]
This looks great! Can I change the stroke color and pass rgba-colors to every traceview?

**Comment by goessner** [comment]
It's easy to do ... so do it by yourself.

1. give view a `stroke` member
2. simply try it out

**Comment by jauhl** [comment]
Unfortunately your models give me a typeerror. I have the latest version (3 days old) of `g2` and just compiled `mec2` from the latest master-branch.
![grafik](https://user-images.githubusercontent.com/19715408/43974397-ba7f0bc2-9cda-11e8-88ab-2ea042122cc5.png)

**Comment by goessner** [comment]
I have sent you A current version of g2 Sitz kg Last Email...


Von meinem Samsung Galaxy Smartphone gesendet.
-------- Ursprüngliche Nachricht --------Von: jauhl <notifications@github.com> Datum: 10.08.18  20:22  (GMT+01:00) An: goessner/mec2 <mec2@noreply.github.com> Cc: Stefan Goessner <stefan@goessner.net>, Author <author@noreply.github.com> Betreff: Re: [goessner/mec2] trace view ... (#7) 
Unfortunately your models give me a typeerror. I have the latest version (3 days old) of g2 and just compiled mec2 from the latest master-branch.



—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or mute the thread.
{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/goessner/mec2","title":"goessner/mec2","subtitle":"GitHub repository","main_image_url":"https://assets-cdn.github.com/images/email/message_cards/header.png","avatar_image_url":"https://assets-cdn.github.com/images/email/message_cards/avatar.png","action":{"name":"Open in GitHub","url":"https://github.com/goessner/mec2"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jauhl in #7: Unfortunately your models give me a typeerror. I have the latest version (3 days old) of `g2` and just compiled `mec2` from the latest master-branch.\r\n![grafik](https://user-images.githubusercontent.com/19715408/43974397-ba7f0bc2-9cda-11e8-88ab-2ea042122cc5.png)\r\n"}],"action":{"name":"View Issue","url":"https://github.com/goessner/mec2/issues/7#issuecomment-412165582"}}}
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/goessner/mec2/issues/7#issuecomment-412165582",
"url": "https://github.com/goessner/mec2/issues/7#issuecomment-412165582",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
},
{
"@type": "MessageCard",
"@context": "http://schema.org/extensions",
"hideOriginalBody": "false",
"originator": "AF6C5A86-E920-430C-9C59-A73278B5EFEB",
"title": "Re: [goessner/mec2] trace view ... (#7)",
"sections": [
{
"text": "",
"activityTitle": "**jauhl**",
"activityImage": "https://assets-cdn.github.com/images/email/message_cards/avatar.png",
"activitySubtitle": "@jauhl",
"facts": [

]
}
],
"potentialAction": [
{
"name": "Add a comment",
"@type": "ActionCard",
"inputs": [
{
"isMultiLine": true,
"@type": "TextInput",
"id": "IssueComment",
"isRequired": false
}
],
"actions": [
{
"name": "Comment",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueComment\",\n\"repositoryFullName\": \"goessner/mec2\",\n\"issueId\": 7,\n\"IssueComment\": \"{{IssueComment.value}}\"\n}"
}
]
},
{
"name": "Close issue",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"IssueClose\",\n\"repositoryFullName\": \"goessner/mec2\",\n\"issueId\": 7\n}"
},
{
"targets": [
{
"os": "default",
"uri": "https://github.com/goessner/mec2/issues/7#issuecomment-412165582"
}
],
"@type": "OpenUri",
"name": "View on GitHub"
},
{
"name": "Unsubscribe",
"@type": "HttpPOST",
"target": "https://api.github.com",
"body": "{\n\"commandName\": \"MuteNotification\",\n\"threadId\": 367119838\n}"
}
],
"themeColor": "26292E"
}
]

**Comment by goessner** [comment]
did you resolve your errors with the new 'g2.js' ?

**Comment by jauhl** [comment]
I did, thank you. The email wasn't forwarded to my gmail yet again. I figured out that this happens with emails containing specific attachment types, including js files, which google refuses to accept (and send). I now set up an imap account for my fh-mail and just got notified for your microapp.js email. So at least this problem should be solved...

I haven't testet the views thoroughly yet, but i couldn't get different stroke-colors to work by simply adding adding a `stroke` property.
